SOURCE/Fiducializing/autoProcessSignal.m

```diff
@@ -367,105 +484,59 @@ function DetectRecovery(newBeatIdx)
+    errordlg(msg)
+    success = 0;
+    return
+end
+
+%%%% get tstart/end  values as saved in the fids
+for tStartIdx=tStartIndeces % loop trought to find global t wave
+    if length(TS{newBeatIdx}.fids(tStartIdx).value) == 1
+        tstart = TS{newBeatIdx}.fids(tStartIdx).value * ones(numchannels,1);
         break
     end
 end
-for tend_idx=tend_indeces % loop trought to find global t wave
-    if length(TS{newBeatIdx}.fids(tend_idx).value) == 1
-        tend = TS{newBeatIdx}.fids(tend_idx).value * ones(numchannels,1);
+for tEndIdx=tEndIndeces % loop trought to find global t wave
+    if length(TS{newBeatIdx}.fids(tEndIdx).value) == 1
+        tend = TS{newBeatIdx}.fids(tEndIdx).value * ones(numchannels,1);
         break
     end
 end
 
 
-
 %%%% sort values
 ts = min([tstart tend],[],2);
 te = max([tstart tend],[],2);
 
 %%%% set up some stuff
 win = ScriptData.RECWIN;
 deg = ScriptData.RECDEG;
-neg = ScriptData.RECNEG;
 
 %%%% get the recovery values for each lead
-for leadNumber=1:numchannels
-    rec(leadNumber) = ARdetect(TS{newBeatIdx}.potvals(leadNumber,ts(leadNumber):te(leadNumber)),win,deg,neg)/ScriptData.SAMPLEFREQ + ts(leadNumber);
+
+[recFktHandle, success] = getRecFunction;
+if ~success, return, end
+try
+    for leadNumber=1:numchannels
+        rec(leadNumber) = recFktHandle(TS{newBeatIdx}.potvals(leadNumber,ts(leadNumber):te(leadNumber)),win,deg)/ScriptData.SAMPLEFREQ + ts(leadNumber);
+    end
+catch
+    errordlg('The selected function used to find the recoveries caused an error. Aborting...')
+    success = 0;
+    return
 end
 
 %%%% put the recovery values in fids
 TS{newBeatIdx}.fids(end+1).type=13;
 TS{newBeatIdx}.fids(end).value=rec;
 
-end
-
-function x = ARdetect(sig,win,deg,pol,ndrange)
-if nargin == 4
-    ndrange = 0;
-end
-
-%%%% if sigdrange to small compared to noisedrange (ndrange), return
-%%%% x=len(sig)
-sigdrange = max(sig)-min(sig);  
-if (sigdrange <= 1.75*ndrange)
-    x = length(sig);
-    return;
-end
-
-%make sure win is uneven
-if mod(win,2) == 0, win = win + 1; end
-
-%%%% return x=1, if len(sig)<win
-if length(sig) < win, x=1; return; end
-
-% Detection of the minimum derivative
-% Use a window of 5 frames and fit a 2nd order polynomial
-
-cen = ceil(win/2);
-X = zeros(win,(deg+1));
-L = [-(cen-1):(cen-1)]';
-for p=1:(deg+1)
-    X(:,p) = L.^((deg+1)-p);
-end
-
-E = (X'*X)\X';
-
-sig = [sig sig(end)*ones(1,cen-1)];
-
-a = filter(E(deg,[win:-1:1]),1,sig);
-dy = a(cen:end);
-
-if pol == 1
-    [mv,mi] = min(dy(cen:end-cen));
-else
-    [mv,mi] = max(dy(cen:end-cen));
-end
-mi = mi(1)+(cen-1);
-
-% preset values for peak detector
-
-win2 = 5;
-deg2 = 2;
-
-cen2 = ceil(win2/2);
-L2 = [-(cen2-1):(cen2-1)]';
-for p=1:(deg2+1), X2(:,p) = L2.^((deg2+1)-p); end
-c = inv(X2'*X2)*X2'*(dy(L2+mi)');
-
-if abs(c(1)) < 100*eps, dx = 0; else dx = -c(2)/(2*c(1)); end
-
-dvdt = 2*c(1)*dx+c(2);
-
-dx = median([-0.5 dx 0.5]);
-
-x = mi+dx-1;
 
+success = 1;
```

Comment:
Consider making this variable name different from all the other times that you use this